cron/jobstats.go

```diff
@@ -0,0 +1,117 @@
+package cron
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+type Job struct {
+	Category	string
+	Status		string
+	Count		int
+}
+
+//jobs/submitted
+func GetSubmittedJobCounts(db *sql.DB, startDay string, endDay string)([]Job, error) {
+	var jobType *string
+	var count int
+	//skip curl wrapper jobs
+	query := `SELECT b.job_type, count(b.*) AS count
+            FROM (
+              SELECT a.id,
+                CASE WHEN array_length(a.types, 1) = 1 THEN a.types[1]
+              ELSE 'Mixed'
+           END AS job_type
+           FROM (
+               SELECT j.id, array_agg(DISTINCT t.name) AS types FROM jobs j
+               JOIN job_steps s ON j.id = s.job_id
+               JOIN job_types t ON s.job_type_id = t.id
+               WHERE j.start_date >= ($1 :: DATE) AND j.start_date <= ($2 :: DATE)
+               AND j.app_id != '1e8f719b-0452-4d39-a2f3-8714793ee3e6'
+               GROUP BY j.id
+           ) AS a
+          ) AS b
+         GROUP BY b.job_type
+         ORDER BY b.job_type;`
+
+	rows, err := db.Query(query, startDay, endDay)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	var jobs []Job
+
+	for i := 0; rows.Next(); i++ {
```

Comment:
Since `rows.Next()` puts out an `EOF` at the end, you don't need to do the `i` variable at all. You should be able to just do `for rows.Next() { ... }`.

Reply:
True, this was left over from before I changed it to append.